### ISSUE TITLE
Create dimensional table for GTFS RT validation rules (e.g. ids and descriptions)

### ISSUE THREAD

**Issue by machow** [cal-itp/data-infra#833]
See https://github.com/cal-itp/data-infra/issues/787#issuecomment-987305649. This will allow us to link our validation roll-ups to rule descriptions / codes / severity levels

* rule_key
* rule_id
* rule_title
* rule_description
* rule_occurrence_suffix

**Comment by lauriemerrell** [comment]
TLDR -- I need to talk @machow about this next week and revisit whether we want to just load this dimensional table from the rules we actually encounter in our validation results. 

Full notes on this:

We thought that because the RT validator outputs have a more consistent structure (always same 5 fields) than the schedule validator outputs (variable fields based on error contents), perhaps we could load a static table of validation rules based on the [validator documentation](https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/RULES.md#table-of-errors). Since one of the concerns here is space (reduce the amount of times we have to store the repeated validation rule contents in each validation output), this seemed potentially preferable to extracting validations that we have encountered (which is [what we do in the GTFS schedule pipeline](https://github.com/cal-itp/data-infra/blob/main/airflow/dags/gtfs_views/validation_dim_codes.sql)). 

However, the validator documentation is not structured (narrative text vs. table) and it does not explicitly document all 5 fields that are actually returned by the validator (it lists rule ID, severity, title, and description within the text but does not list occurrence suffix). The validator itself also doesn't expose a way to access the rule list as far as I can tell. 

I thought that perhaps I could scrape the [raw Java code ](https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/gtfs-realtime-validator-lib/src/main/java/edu/usf/cutr/gtfsrtvalidator/lib/validation/ValidationRules.java). I talked this option over pretty extensively with @mjumbewu and I have an initial implementation ready. With this approach, we would:
* Scrape the raw Java code to get the validation rules (using a one-off script located in the gtfs-rt-validator-api repo) 
* Write the codes out to a CSV (also located in gtfs-rt-validator-api) 
* Create a wrapper in gtfs-rt-validator-api that loads that CSV and exposes it to the Airflow DAGs over here in data-infra
* Load the CSV contents into a view via DAG

However, the snag on this relates to versioning. 

I am not able to tell whether the JAR we're using to get the validator is actually a fixed version. It seems from their documentation like that JAR may be updated dynamically, and the validation rules have been updated occasionally (most recently in 2020). 

Therefore, it seems like:
* There could be rule updates that would be picked up automatically in the JAR we're using, so our scraped CSV could get out of sync with the results we're actually getting from the validator 
* I can't tell which version of the rules we're currently using (I can't figure out when the JAR we're currently using is from -- whether it's the original 2018 release or has more recent changes) 
* Even if we use a slowly changing dimension table , what we would set as the active dates seems ambiguous since the validator versioning is opaque -- it will be hard to keep any scraped list in sync with the actual validation results because code changes in the validator repo may or may not be making it into the actual validator that we are using

So, I think we may need to revisit just extracting the validations that we actually see in our data to put into this table. But I think that will then make this ticket dependent on #902 because it will relate to how exactly we are storing the validation results.

**Comment by lauriemerrell** [comment]
Further notes:
* Talked to Kevin today, he said that we might be able to get the validator version with a HTTP GET request. So that might be one way to approach the versioning to understand when the JAR is updated.
* Maybe related question -- do we want to consider making a "local" copy of the JAR and not pointing at their S3 location that seems like it can get updated dynamically?

**Comment by lauriemerrell** [comment]
Chatted with @atvaccaro about this:
* We do want to take a cut of their JAR in our validator repo 
* Also will put a CSV of the rules there
* Validator API will expose the contents of the CSV
